rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -933,8 +940,54 @@ void main_window::HandlePackageInstallation(QStringList file_paths)
 
 			if (i == (count - 1))
 			{
+				pdlg.setHidden(true);
 				m_gui_settings->ShowInfoBox(tr("Success!"), tr("Successfully installed software from package(s)!"), gui::ib_pkg_success, this);
 			}
+
+			if (std::string game_dir = fs::get_parent_dir(bootable_path, 2); fs::is_dir(game_dir) && fs::is_file(bootable_path))
+			{
+				const std::string target_cli_args = fmt::format("--no-gui \"%%RPCS3_GAMEID%%:%s\"", game_dir.substr(game_dir.find_last_of(fs::delim) + 1));
+				const std::string target_icon_dir = fmt::format("%sIcons/game_icons/%s/", fs::get_config_dir(), sstr(package.title_id));
+				const std::string icon_path =  bootable_path + "/../../ICON0.PNG";
+
+#ifdef _WIN32
+				if (gui::utils::create_shortcut(sstr(package.title), target_cli_args, sstr(package.title), icon_path, target_icon_dir, gui::utils::shortcut_location::rpcs3_shortcuts))
+				{
+					gui_log.success("Created a shortcut for '%s' at '%s/games/shortcuts/'", sstr(package.title.simplified()), fs::get_config_dir());
+				}
+#endif
+
+				std::initializer_list<std::pair<std::string_view, gui::utils::shortcut_location>> installing_locations =
+				{
+					{"desktop"sv, gui::utils::shortcut_location::desktop},
+#ifdef _WIN32
+					{"start menu"sv, gui::utils::shortcut_location::applications},
+#elif defined(__APPLE__)
+					{"dock"sv, gui::utils::shortcut_location::applications},
+#else
+					{"launcher"sv, gui::utils::shortcut_location::applications},
+#endif
+
+				};
+
+				for (const auto& [location_string, location] : installing_locations)
+				{
+					if (QMessageBox::question(this, tr("Install shortcut to %1?").arg(package.title), tr("Would you like to install a shortcut to the application on the %1?").arg(qstr(location_string.data())),
+							QMessageBox::Yes | QMessageBox::No, QMessageBox::No) == QMessageBox::Yes)
+					{
+						if (gui::utils::create_shortcut(sstr(package.title), target_cli_args, sstr(package.title), icon_path, target_icon_dir, location))
+						{
+							gui_log.success("Created %s shortcut for %s", location_string, sstr(package.title.simplified()));
+							QMessageBox::information(this, tr("Success!"), tr("Successfully created a shortcut."));
+						}
+						else
+						{
+							gui_log.error("Failed to create %s shortcut for %s", location_string, sstr(package.title.simplified()));
+							QMessageBox::warning(this, tr("Warning!"), tr("Failed to create a shortcut!"));
+						}
+					}
+				}
+			}
```

Comment:
why is this asking 2 times?
just create a list of checkboxes or whatever.
Also, what happens if i install 20 files and go to the toilet?
Will I come back to some annyoing popup and only one file was installed?

Reply:
I see, need to ask before the installation of any of them.

Reply:
Fixed. (asks once later)